disco_aws_automation/disco_ssm.py

```diff
@@ -0,0 +1,175 @@
+"""
+Manage AWS SSM document creation and execution
+"""
+import os
+import logging
+import time
+import json
+
+import boto3
+
+from botocore.exceptions import ClientError
+from . import read_config
+from .resource_helper import throttled_call, wait_for_state_boto3
+from .exceptions import TimeoutError
+
+logger = logging.getLogger(__name__)
+
+
+SSM_DOCUMENTS_DIR = "ssm/documents"
+SSM_EXT = ".ssm"
+SSM_WAIT_TIMEOUT = 5 * 60
+SSM_WAIT_SLEEP_INTERVAL = 15
+DOCUMENT_PREFIX = "asiaq-"
```

Comment:
My impression is that filtering the AWS documents is definitely a reason. The other reason is to provide some kind of grouping for the documents.

I was also wondering if we could restrict certain documents from being run against certain instances, something like: the documents from Product Services group can only be run against instances with PS being the product line. I am not sure if this is possible though.